docs/DafnyRef/Types.md

```diff
@@ -1705,29 +1776,32 @@ such a `C` object can be used as a value of type `J`.
 
 ## 13.1. Type `object`
 ````grammar
-ObjectType_ = "object"
+ObjectType_ = "object" | "object?"
 ````
 
-There is a built-in trait `object` that is a supertype of all
-reference types.
+There is a built-in trait `object?` that is a supertype of all
+reference types and a trait `object` that is a supertype of all non-null reference types..
```

Comment:
Technically, `object` is not a trait but a subset type. However, I like your sentence, because I think it says something useful and more easily understood.

Reply:
OK